main.cpp

```diff
@@ -2,11 +2,11 @@
 #include <iostream>
 
 int main() {
-    Image image;
-    image.rows = 512;
-    image.columns = 512;
+    Image* image = new Image();
+    image->rows = 512;
+    image->columns = 512;
 	std::cout << "Brightening a 512 x 512 image\n";
-    ImageBrightener brightener(image);
+    ImageBrightener brightener(*image);
```

Comment:
Typically, passing around non-const references reduces transparency in code-flow.
In this case:
The pixels inside the image instance start out by being 'not bright'. Once the `ImageBrightener` is done, the same instance has a brightened image. If this image needs to be passed to routines requiring an un-brightened image, we need to pass it carefully in the sequence of code. It's even more tricky to maintain when the sequence becomes more complex with new features.
Perhaps it's because a variable called `image` starts by being the original image and transforms to become bright somewhere in the middle. What do you think?

Reply:
Indeed, that's a fair point. I could even go as far as 
```suggestion
    bool is_image_brightended = ImageBrightener brightener(/*const */ *original_image, &brightened_image);
```